core/src/main/scala/org/apache/spark/sql/crossdata/execution/datasources/ddl.scala

```diff
@@ -243,6 +243,22 @@ private[crossdata] case class AddJar(jarPath: String)
   }
 }
 
+private[crossdata] case class AddApp(xdContext:XDContext, jarPath: String, className:String,alias:Option[String]=None)
+  extends LogicalPlan with RunnableCommand {
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    if (jarPath.toLowerCase.startsWith("hdfs://") || File(jarPath).exists) {
+      sqlContext.sparkContext.addJar(jarPath)
+    } else if (File(jarPath).exists){
+      xdContext.addJar(jarPath)
```

Comment:
This is unreachable code. Whenever `File(jarPath).exists` is true it will also be `jarPath.toLowerCase.startsWith("hdfs://") || File(jarPath).exists`